provider/common/bootstrap.go

```diff
@@ -200,6 +201,31 @@ func BootstrapInstance(ctx environs.BootstrapContext, env environs.Environ, args
 	return result, selectedSeries, finalize, nil
 }
 
+func formatHardware(hw *instance.HardwareCharacteristics) string {
+	if hw == nil {
+		return ""
+	}
+	out := make([]string, 0, 3)
+	if hw.Arch != nil && *hw.Arch != "" {
+		out = append(out, *hw.Arch)
+	}
+	if hw.Mem != nil && *hw.Mem > 0 {
+		out = append(out, fmt.Sprintf("mem=%s", formatMemory(*hw.Mem)))
+	}
+	if hw.CpuCores != nil && *hw.CpuCores > 0 {
+		out = append(out, fmt.Sprintf("cores=%d", *hw.CpuCores))
+	}
+	return strings.Join(out, " ")
+}
+
+func formatMemory(m uint64) string {
```

Comment:
I think this code seems very familiar.
Check here: https://github.com/juju/juju/blob/master/instance/instance.go#L251

If we are here, instead of writing the same semantic functions to parse memory why not make one global one(move it to utils or smth) and use it in the entire codebase?

What's your thoughts?